File: benchmarking/pure.rb
require 'statsig'
require 'benchmark'

secret = ENV['test_api_key']
Statsig.initialize(secret)

user = StatsigUser.new({'userID' => 'Dan'})

time = Benchmark.measure {
    init_res = {}
    1000.times do
        init_res = Statsig.get_client_initialize_response(user)

    end 
    puts "Client init res: #{init_res}"
}

puts (time.real * 1000) 

File: benchmarking/pure.py
import os

from time import perf_counter
from statsig import statsig, StatsigUser

statsig_secret = os.environ.get('test_api_key')
statsig.initialize(statsig_secret)

user = StatsigUser("Dan")

start = perf_counter()
for lp in range(1000):
    result = statsig.get_client_initialize_response(user)
end = perf_counter()

print(result)
print(f"Duration: {(end - start) * 1000}")

